lib/sda/context.go

```diff
@@ -29,15 +30,36 @@ func newDefaultContext(h *Host, o *Overlay, servID ServiceID) *defaultContext {
 	}
 }
 
+// NewTreeNodeInstance implements the Context interface method
+func (dc *defaultContext) NewTreeNodeInstance(t *Tree, tn *TreeNode, protoName string) *TreeNodeInstance {
+	return dc.Overlay.NewTreeNodeInstanceFromService(t, tn, ProtocolNameToID(protoName), dc.servID)
+}
+
+// SendRaw sends a message to the entity
+func (dc *defaultContext) SendRaw(e *network.Entity, msg interface{}) error {
+	return dc.Host.SendRaw(e, msg)
+}
+
+// Entity returns the entity the service uses
 func (dc *defaultContext) Entity() *network.Entity {
 	return dc.Host.Entity
 }
 
-// NewTreeNodeInstance implements the Context interface method
-func (dc *defaultContext) NewTreeNodeInstance(t *Tree, tn *TreeNode) *TreeNodeInstance {
-	return dc.Overlay.NewTreeNodeInstanceFromService(t, tn, dc.servID)
+// GetID returns the service-id
+func (dc *defaultContext) GetID() ServiceID {
+	return dc.servID
 }
 
-func (dc *defaultContext) SendRaw(e *network.Entity, msg interface{}) error {
-	return dc.Host.SendRaw(e, msg)
+// CreateProtocol makes a TreeNodeInstance from the root-node of the tree and
+// prepares for a 'name'-protocol. The ProtocolInstance has to be added later.
+func (dc *defaultContext) CreateProtocol(t *Tree, name string) (ProtocolInstance, error) {
+	pi, err := dc.Overlay.CreateProtocolService(dc.servID, t, name)
+	return pi, err
+}
+
+// CreateProtocolAuto is like CreateProtocol but doesn't bind a service to it,
+// so it will be handled automatically by the SDA.
+func (dc *defaultContext) CreateProtocolAuto(t *Tree, name string) (ProtocolInstance, error) {
```

Comment:
Mhhh. that's becoming confusing. 
* Why a `Context` should need to create a protocol not tied to a `Service` (using `CreateProtocolAuto`) when a `Context` is *only* used by  a `Service` ? We only create protocols without a service in our tests & simulations.

* I find it confusing to have both but we did not find any better way for the moment so it's fine; but maybe we could be more helpful with a more precise naming ? `CreateProtocolService` or `CreateServiceProtocol` // and `CreateProtocolIndependant` or `CreateIndependantProtocol` ? I don't know, I'm not sure, but both are very similar and it's not easy to remember which one is the right one to use just by looking at the name `CreateProtocolAuto`